functional_data_generator/lib/builder.dart

```diff
@@ -85,17 +81,20 @@ String _generateDataType(Element element) {
   return '$dataClass $lensesClass';
 }
 
-String _typeAsCode(DartType type) {
-  if (type is FunctionType) {
-    final positionArgs = type.parameters.where((p) => p.isPositional).map((p) => _typeAsCode(p.type)).join(', ');
-    final namedArgs = type.parameters.where((p) => p.isNamed).map((p) => '${_typeAsCode(p.type)} ${p.name}').join(', ');
-
-    final parameters = namedArgs.isEmpty ? positionArgs : positionArgs + ', {$namedArgs}';
+TypeAnnotation _findTypeNode(FieldElement element) {
+  final parsedLibrary =
+      element.session.getParsedLibraryByElement(element.library);
+  final declaration = parsedLibrary.getElementDeclaration(element);
+  return ((declaration.node as VariableDeclaration).parent
+          as VariableDeclarationList)
+      .type;
+}
 
-    return '${_typeAsCode(type.returnType)} Function($parameters)';
-  } else {
-    return type.displayName;
+String _typeAsCode(TypeAnnotation type) {
+  if (type == null) {
+    return 'dynamic';
   }
+  return type.toString();
```

Comment:
This is equivalent to `type.displayName` and the `displayName` documentation says:

> Clients should not depend on the content of the returned value

In `sum_types` I reconstruct the type name "manually" [like this](https://github.com/werediver/sum_types.dart/blob/0040266f788db20564887e9d7f6d8e8ae54f0588/sum_types_generator/lib/src/sum_type_spec.dart#L159).